packages/components/Callout/macos/CalloutWindow.swift

```diff
@@ -22,11 +22,29 @@ class CalloutWindow: NSWindow {
 		// Dismiss the Callout if the window is no longer active.
 		NotificationCenter.default.addObserver(self, selector: #selector(dismissCallout), name: NSApplication.didResignActiveNotification, object: nil)
 
-		// Dismiss the Callout if the user touched/clicked outside the bounds of the callout.
+		// Dismiss the Callout if the user clicked outside the Callout Window or any of our child windows.
 		mouseEventMonitor = NSEvent.addLocalMonitorForEvents(matching: .leftMouseUp, handler: { [weak self] (event) -> NSEvent? in
-			if (event.window != self) {
+			guard let clickedWindow = event.window else {
+				return event
+			}
+
+			var shouldDismissCallout = true
+
+			if (clickedWindow.isEqual(to: self) ) {
+				// The click happened in our own window. 
+				shouldDismissCallout = false
+			} else if (clickedWindow.isEqual(to: self?.parent) && self?.parent is CalloutWindow) {
```

Comment:
again super nit but would this make more swift like? 
 if let parentWindw = self?.parent as? CalloutWindow, clickedwindow.isequal(to: parentWindow).

Reply:
`if let` is usually used to resolve an optional, and in this case we never use the resulting `parentWindow` variable. 
Maybe something like this?
``` 
} else if (clickedWindow.isEqual(to: self?.parent as? CalloutWindow)) {
```

Reply:
https://developer.apple.com/swift/blog/?id=23 This blog is telling me that if the conversion`self?.parent as? CalloutWindow` fails, it evaluates to nil, so I think this is OK?

> let animal: Animal = Cat()
animal as? Dog	// evaluates to nil
animal as! Dog	// triggers a runtime error

Reply:
An aside... I think this logic only works for one level of submenus. If someone had 2 levels of submenus (Menu -> Submenu -> Submenu) I think it breaks :/